txn_writer.go

```diff
@@ -349,6 +361,13 @@ func (sw *SafeWriter) Write(root string, sm gps.SourceManager) error {
 		os.RemoveAll(vendorbak)
 	}
 
+	if len(sw.Payload.Manifest.Dependencies) == 0 {
```

Comment:
I looked into the nil pointer error from the build and it is because the `Write` function doesn't _always_ have a manifest given to it, so `sw.Payload.Manifest` was `nil`. 

If you move this check under the `sw.Payload.HasManifest()` check above (https://github.com/golang/dep/blob/master/txn_writer.go#L254), then it will only append the help text when we have a manifest file to write.